CRO i.2: integrate Jetpack Product Slide Out Card capability

elliottprogrammer opened the pull request:
### Changes proposed in this Pull Request

This PR makes it possible to have the Jetpack Product Slide Out Cards in the features section of the 3 Plans.

#### Notes:
This PR is a branch off of `add/product-card-i2-features`, [#46789](https://github.com/Automattic/wp-calypso/pull/46789) .
This working branch did not yet have the JetpackProductSlideOutCard component available yet, so in it's place is testing/placeholder markup displaying all the component values and props, whereas the `JetpackProductSlideOutCard` component and import statement is there, but it is commented out. Once this PR is merged, we can create another PR to remove the placeholder markup, and uncomment out the component markup, making it active.

This PR also does not handle how the product slide-out cards are displayed if/when products are already owned.
This PR is mainly just about making it possible to drop in the JetpackProductSlideOutCard component in the Features section and make it functional. There may be some slight design improvements that can be done in a following PR when we activate the JetpackProductSlideOutCard component.

#### Implementation Notes:

 - Created 2 constants (`FEATURE_TO_PRODUCT_ALT_V2`, `FEATURE_TO_MONTHLY_PRODUCT_ALT_V2`) to map the product features to the appropriate product slug.
 - Had to pass the product `billingTerm` and `onProductClick` prop through the `JetpackProductCardAlt2` component and into the `JetpackProductCardFeatures` component.
 - Created a special `constants.FEATURE_PRODUCT_SCAN_V2_NO_SLIDEOUT` feature key in order to be able to specifically exclude it from rendering the product slide-out card only in the Jetpack Security Real-time plan. (Instead of the slide-out card, it renders the info-popout).
- Modified the `JetpackProductCardFeaturesItem` component to render and toggle the "Learn more" & "Close" product dropdown toggle.
 - Created a new `FeaturesProductSlideOut` component containing the logic and rendering of the new `JetpackProductSlideOutCard` component.

### Testing instructions

The link to the Figma design mockup and prototype is located here in this Asana card:
1196341175636977-as-1198496807750941/f

1. Run this PR (both Calypso and Cloud)
2. Visit the `Plans" page in Calypso and the `/pricing` page in Jetpack Cloud.
3. Select a Jetpack site with Free Plan (No plans owned)
4. Expand all the Plans features sections.
5. Click to open each slide-out product and inspect the product info(props) and that it is correct for each product. Check both Monthly and Yearly.
6. Verify the "Learn more" and "Close" dropdown functionality works for all of them.
7. Verify that the "Scan" product for Jetpack Security Real-time does **_not_** have a product slide-out but instead an Info pop-out.

Screenshot:
![Markup 2020-10-27 at 20 39 20](https://user-images.githubusercontent.com/11078128/97388106-a1942880-1894-11eb-80db-049215295aca.png)


Fixes: 1196341175636977-as-1198930356649981/f

monsieur-z commented on client/components/jetpack/card/jetpack-product-card-alt-2/features-item.tsx:
```diff
@@ -1,49 +1,115 @@
+
+	const slideOutProduct = slugToSelectorProduct( slideOutProductSlug );
+
+	const toggleSlideOut = () => {
```
Just a suggestion here: since `toggleSlideOut` is passed as a prop to `Button`, we could wrap it with `useCallback`, so that a new instance isn't created at each render. It wouldn't change anything here I believe, but I'd say it's good practice since it could potentially prevent re-renders (for memoized components).

elliottprogrammer replied:
Yes, great suggestion, thank you 🙂

monsieur-z commented:
Looks good to me @elliottprogrammer! I haven't approved it just yet since master has been updated recently and there are some conflicts.

elliottprogrammer commented:
> Looks good to me @elliottprogrammer! I haven't approved it just yet since master has been updated recently and there are some conflicts.

Awesome @monsieur-z , I rebased, and now added the `JetpackProductSlideOutCard` component (and added the `useEffect()`).  It looks like some of the styles of the main product card are overriding some styles of the `JetpackProductSlideOutCard`. This can be corrected in upcoming PR (as well as some other tasks like handling owned products, and making sure the slide-out product CTA's are checking out properly).  If you don't mind briefly reviewing again and let me know what you think. 🙂

monsieur-z commented on client/components/jetpack/card/jetpack-product-card-alt-2/features-item.tsx:
```diff
@@ -3,32 +3,85 @@
+
+	const toggleSlideOut = useCallback( () => {
+		setSlideOutExpanded( ( state ) => ! state );
+	}, [] );
```
Just need to set `setSlideOutExpanded` to get rid of the linter warning :)

monsieur-z reviewed (approved):
LGTM! 👍

wp-desktop reviewed (changes requested):
WordPress Desktop CI Failure for job "wp-desktop-mac".

@elliottprogrammer please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/936442). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5062107

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5062107).
